### PR TITLE
boards: arm: stm32_min_dev: Add ADC 1

### PR DESCRIPTION
**Pull request by gon1332**
Added ADC_1

### CODE REVIEW

File: boards/arm/stm32_min_dev/pinmux.c [hunk 1]
```diff
@@ -57,6 +57,9 @@ static const struct pin_config pinconf[] = {
 	{STM32_PIN_PA11, STM32F1_PINMUX_FUNC_PA11_USB_DM},
 	{STM32_PIN_PA12, STM32F1_PINMUX_FUNC_PA12_USB_DP},
 #endif /* CONFIG_USB_DC_STM32 */
+#ifdef CONFIG_ADC_1
```
Comment (galak) [inline boards/arm/stm32_min_dev/pinmux.c:60]: Use the nodelabel form for this as we are migrating towards that:

`#if DT_HAS_NODE(DT_NODELABEL(adc1))`

**Review by galak — CHANGES_REQUESTED** [review]
change pinmux to use nodelabel.

File: boards/arm/stm32_min_dev/pinmux.c (hunk 1 above)
Comment (gon1332) [reply boards/arm/stm32_min_dev/pinmux.c]: Done. Could you point me to the conversation on this change?

**Comment by gon1332** [comment]
After trying to support ADC_2 a conflict occurred. ADC_1 and ADC_2 as the Reference Manual of STM32F103xx states (pg. 199, table 61.), share in the same IRQ table position (18).

I've added the extra adc2 node needed along with adc1 in the according soc .dts but during the building there is this parsing error:
```
FAILED: zephyr/isr_tables.c
cd /home/gon1332/Development/zephyrproject/build/zephyr && /home/gon1332/zephyr-sdk/arm-zephyr-eabi/bin/arm-zephyr-eabi-objcopy --input-target=elf32-littlearm --output-target=binary --only-section=.intList /home/gon1332/Development/zephyrproject/build/zephyr/zephyr_prebuilt.elf isrList.bin && /usr/bin/python3.6 /home/gon1332/Development/zephyrproject/zephyr/arch/common/gen_isr_tables.py --output-source isr_tables.c --kernel /home/gon1332/Development/zephyrproject/build/zephyr/zephyr_prebuilt.elf --intlist isrList.bin   --sw-isr-table --vector-table
gen_isr_tables.py: error: multiple registrations at table_index 18 for irq 18 (0x12)
Existing handler 0x80026e9, new handler 0x80026e9
Has IRQ_CONNECT or IRQ_DIRECT_CONNECT accidentally been invoked on the same irq multiple times?
```

Could you point me out on how to overcome this? I've uploaded the changes in a new commit.

**Comment by erwango** [comment]
@gon1332,  Use in parallel of 2 ADC instances sharing the same IRQ is not yet supported.
cf https://github.com/zephyrproject-rtos/zephyr/pull/24269/files#r408644437 for a workaround

**Comment by gon1332** [comment]
> @gon1332, Use in parallel of 2 ADC instances sharing the same IRQ is not yet supported.
> cf https://github.com/zephyrproject-rtos/zephyr/pull/24269/files#r408644437 for a workaround

Removed the extra commit about ADC_2 and force-pushed.

File: soc/arm/st_stm32/stm32f1/dts_fixup.h [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 /* SoC level DTS fixup file */
 
-#define DT_ADC_1_NAME			DT_ST_STM32_ADC_40012400_LABEL
+#define DT_ADC_0_NAME			DT_LABEL(DT_INST(0, st_stm32_adc))
```
Comment (erwango) [inline soc/arm/st_stm32/stm32f1/dts_fixup.h:5]: This should not be required

**Review by erwango — CHANGES_REQUESTED** [review]

File: soc/arm/st_stm32/stm32f1/dts_fixup.h (hunk 2 above)
Comment (gon1332) [reply soc/arm/st_stm32/stm32f1/dts_fixup.h]: Done

File: soc/arm/st_stm32/stm32f1/dts_fixup.h [hunk 3]
```diff
@@ -2,8 +2,6 @@
 
 /* SoC level DTS fixup file */
 
-#define DT_ADC_1_NAME			DT_ST_STM32_ADC_40012400_LABEL
```
Comment (erwango) [inline soc/arm/st_stm32/stm32f1/dts_fixup.h:5]: I'm fine removing this if not needed, but for coherency, I prefer it is removed in all STM32 series as part of a dedicated PR

**Review by erwango — CHANGES_REQUESTED** [review]
Otherwise, lgtm.

File: soc/arm/st_stm32/stm32f1/dts_fixup.h (hunk 3 above)
Comment (gon1332) [reply soc/arm/st_stm32/stm32f1/dts_fixup.h:5]: I could do it.

**Review by erwango — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

File: boards/arm/stm32_min_dev/pinmux.c [hunk 4]
```diff
@@ -57,6 +57,9 @@ static const struct pin_config pinconf[] = {
 	{STM32_PIN_PA11, STM32F1_PINMUX_FUNC_PA11_USB_DM},
 	{STM32_PIN_PA12, STM32F1_PINMUX_FUNC_PA12_USB_DP},
 #endif /* CONFIG_USB_DC_STM32 */
+#if DT_HAS_NODE(DT_NODELABEL(adc1))
```
Comment (erwango) [inline boards/arm/stm32_min_dev/pinmux.c:60]: DT_HAS_NODE_STATUS_OKAY

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/stm32_min_dev/pinmux.c (hunk 4 above)
Comment (gon1332) [reply boards/arm/stm32_min_dev/pinmux.c]: Done

**Review by erwango — APPROVED** [review]

**Comment by erwango** [comment]
@gon1332 can you rebase?

**Comment by galak** [comment]
rebased and fixed merge conflict.

**Review by galak — APPROVED** [review]

**Pull request merged**